cli/pact.js

```diff
@@ -0,0 +1,118 @@
+/*jslint node: true */
+'use strict';
+
+/**
+ * Spawns the skyux pact command.
+ * @name pact
+ */
+function pact(command, argv) {
+  const logger = require('../utils/logger');
+  const Server = require('karma').Server;
+  const tsLinter = require('./utils/ts-linter');
+  const skyPagesConfigUtil = require('../config/sky-pages/sky-pages.config');
+  let skyPagesConfig = skyPagesConfigUtil.getSkyPagesConfig(command);
+  let http = require('http');
+  let httpProxy = require('http-proxy');
+  const portfinder = require('portfinder');
+  const url = require('url');
+  const pactServers = require('../utils/pact-servers');
+
+  argv = argv || process.argv;
+  argv.command = command;
+
+  let lintResult;
+
+  const onRunStart = () => {
+    lintResult = tsLinter.lintSync();
+  };
+
+  const onRunComplete = () => {
+    if (lintResult.exitCode > 0) {
+      // Pull the logger out of the execution stream to let it print
+      // after karma's coverage reporter.
+      setTimeout(() => {
+        logger.error('Process failed due to linting errors:');
+        lintResult.errors.forEach(error => logger.error(error));
+      }, 10);
+    }
+  };
+
+  const onExit = (exitCode) => {
+    if (exitCode === 0) {
+      exitCode = lintResult.exitCode;
+    }
+
+    logger.info(`Karma has exited with ${exitCode}.`);
+    process.exit(exitCode);
+  };
+
+  let pactPortPromises = [];
+  // get a free port for every config entry, plus one for the proxy
+  for (let i = 0; i < skyPagesConfig.skyux.pactConfig.pacts.length + 1; i++) {
```

Comment:
You'll need to make sure that `skyPagesConfig.skyux.pactConfig.pacts` exists before trying to use it.

Reply:
:+1: